src/Table.php

```diff
@@ -372,44 +336,13 @@ public function renderView()
         $rows = 0;
         foreach ($this->model as $this->current_id => $tmp) {
             $this->current_row = $this->model->get();
+            $this->hook('beforeRow');
```

Comment:
I think hook should be after updateTotals(), because if you change model values (like in example where you add row after current one), then totals will be messed up too.
On the other hand - sometimes it can be useful to add some additional conditions on updateTotals().

Also i think it would be great if we add ability to return `false` from beforeRow hook and in that case row will not be rendered (and not added to totals) at all. In that case it is good to have hook before updateTotals().